### DISCUSSION TITLE
7DtD Linux Server: Running, but can't connect.

### DISCUSSION THREAD

**Discussion by LordSith77** [GameServerManagers/LinuxGSM#3614] [Q&A]
I have searched, Googled, and everything I could to try and figure this out.

Ubuntu 20.04, running on a HPE ProLiant DL380 G7 Server.:
CPU
Model:      Intel(R) Xeon(R) CPU           X5660  @ 2.80GHz
Cores:      24

Memory
Mem:       total  used   free   cached  available
Physical:  44GB   6.9GB  36GB   7.1GB   36GB
Swap:      2.0GB  72MB   2.0GB

Storage
Filesystem:  /dev/sda5
Total:       941G
Used:        127G
Available:   766G

All needed ports are opened in UFW and forwarded in router.:
Game              26900  udp       1
Game+2         26902  udp       2
Query             26900  tcp       1
Web Admin    5000   tcp       1
Telnet             8081   tcp       0

Server starts, and I can log in to the Web Control Panel. ./sdtdserver details shows it both started and listed.

But it doesn't show up anywhere. Not in Steam servers list, not on the LAN tab, and if I try to connect manually with external or internal IP, it times out.

When I first set it up using regular SteamCMD it worked for 1 day, then gone. I have tried everything under the sun to get this working. So I'm hoping someone here can point me in the right direction. GSM seems to make things easier, but still at a loss as to why it's not showing up anywhere and I can't connect to it. 

Thanks for any and all help.

**Comment by kverb** [comment]
Did you find any solution to this? I'm having the same or similar issue on ubuntu. I can start the game using GSM or just from the command line, and it appears to run; no crashes. But `netstat -tulpn` doesn't show the configured ports as listening.

**Comment by LordSith77** [comment]
Sadly no. I've given up. I've tried every game server for Linux I could find and I couldn't get a single one working.

**Comment by kverb** [comment]
Thanks, and sorry to hear you're still having trouble. I managed to resolve my issue which appears to be a different problem. But in case it helps future visitors, here are some more details about my problem:

- fresh install of Ubuntu 21 in an unprivileged LXC (proxmox host)
- fresh install of LinuxGSM sdtdserver
- update game to 'latest_beta' (alpha 20 at the time)
- migrate an old game save + generated world (I now know that this was incorrectly exported and thus incorrectly imported)
- when i would start the new server, it appeared to start fine (no catastrophic crashing or error messages in console output). But, having never adminstered a 7dtd server before, I failed to realize that the game simply wasn't starting correctly / fully. 
- `netstat -tulpn` and `./sdtdserver monitor` showed that the game's process never registered to listen on any ports. 
- start a default game config with no pre-existing save and everything works. Checking the console log revealed that there should be more happening. It's unfortunate that the game's/GSM's console output doesn't really show anything helpful when the game can't complete starting because of an invalid game save. 

problem with my game save:
In alpha 20, the game state files are in a 'Saves' directory, and when using generated worlds, a 'GenerationData' directory (which corresponds to the `UserDataFolder` property in the server config). My previous export didn't have the 'GenerationData' and would just keep the game process in a limbo state. Bringing the 'GenerationData' directory over to this server fixed everything. Game starts and listens on the specified ports, and the game state and world is brought back to the save.

**Reply by XBadPlayerX** [reply]
I wouldn't use Ubuntu 21 yet.
Use a LTS version.

**Comment by KeepSmileHope** [comment]
Hello there

Sorry for my late answer, there are Couple things to do to get your Server running.

First of all, is it installing properly no dependencies issues ?
Have you forwarded all port you need?

Then here is a tricky part, the normal settings kicking you out from connecting, seach for this line
`<property name="ServerDisabledNetworkProtocols"		value="SteamNetworking"/>`
delete the SteamNetworking and leave the option black like this
`<property name="ServerDisabledNetworkProtocols"		value=""/>`

Some more issues i run in in the couple last weeks:
1) After reinstalling the server, delete ur downloaded world, if its running on same name, and ofcourse ur world you played before.
2) If you playing with mods, leave a comment with stand still 1 Min after logging out, this causes some Corruption on Player data with the Player. If you run straight into logout. <-- Happens twice with mods, after standing still its gone.


Hope this helps.

**Comment by XBadPlayerX** [comment]
I know this is late, but...

It doesn't seem like the ports you have opened are correct.
Here is a link to what you need to have open for ports.
https://7dtd.illy.bz/wiki/Ports

You should have 8gb of swap also.

**Reply by MWRtv** [reply]
From the link [https://7dtd.illy.bz/wiki/Ports](https://7dtd.illy.bz/wiki/Ports) posted here, what does it mean here when it says **\<yourServerIp\>**, (does that mean local IP address or Public IP address):

iptables -A OUTPUT -s **\<yourServerIp\>**/32 -p udp --dport 27000:27099 -m state --state NEW,RELATED,ESTABLISHED -j ACCEPT

iptables -A INPUT  -d **\<yourServerIp\>**/32 -p udp --sport 27000:27099 -m state --state RELATED,ESTABLISHED     -j ACCEPT

iptables -A INPUT  -d **\<yourServerIp\>**/32 -p udp --dport 26900:26903 -m state --state NEW,RELATED,ESTABLISHED -j ACCEPT

iptables -A OUTPUT -s **\<yourServerIp\>**/32 -p udp --sport 26900:26903 -m state --state RELATED,ESTABLISHED     -j ACCEPT

iptables -A INPUT  -d **\<yourServerIp\>**/32 -p tcp --dport 26900 -m state --state NEW,RELATED,ESTABLISHED -j ACCEPT

iptables -A OUTPUT -s **\<yourServerIp\>**/32 -p tcp --sport 26900 -m state --state RELATED,ESTABLISHED     -j ACCEPT

In terms of the 8GB swap needed, do I need to reinstall the OS in order to make a bigger swap? Or can I modify the partitions to make it? Can you guide me in the right direction. I'm trying to get the 7dtd server working, but IDK where I went wrong. I don't even know if this is the right forum to write, if I wanted to make a new topic for help with my situation.

I used SteamCMD to install the 7dtd server files. Edited the serverconfig.xml file to my taste. I used `screen` to start the server with the `./startserver.sh -configfile=serverconfig.xml` command (with no error, but it hangs at the end. I don't know if it's supposed to do that). I opened the necessary ports using `ufw`, (i'm not sure if its necessary to do the `iptables` commands as well at this point). I started the game on my main desktop PC and tried to connect to the **public IP address** with the correct password, but it doesn't work. Btw, I edited (bold and italics) the `./startserver.sh` file to have: 

``./7DaysToDieServer.x86_64 ``**_--configfile=serverconfig.xml_**`` $SERVERDIR/7DaysToDieServer_Data/output_log__`date +%Y-%m-%d__%H-%M-%S`.txt -quit -batchmode -nographics -dedicated $PARAMS``

at the bottom of the file, but I reverted it back because it was verbosing the server log file to the terminal instead of to an actual log file.

I can give more info if needed, but I don't know what else to add right now. Any help would be greatly appreciated.

**Reply by MWRtv** [reply]
I just used my local IP and password as the input for the server in-game and it connected from another computer, so I know my server is working. Only issue is how do I test if external connection is possible. Is there a way to test this from my internal network, or do I have to ask a friend to try to connect. (I made the server unlisted, but if I make it friend invite only would my friend be able to connect to my server, while I'm connected locally and invite him (with the current computer settings)? That would be the next thing I should test.

**Reply by MWRtv** [reply]
One thing I changed what the SteamNetworking option in the serverconfig.xml to just be blank "". I don't know if that would have affected me joining locally or not. (Just thought I'd mention it.)

**Reply by LordSith77** [reply]
As for your serverIP, you need to input your external IP or people outside of your home network won't be able to join.
As for the IPTables, that depends on your Linux version and if it's using IPTables.
As for the swap, you can change it again, depending on what flavor your running. I'm using Ubuntu 20.04, and followed this guide to increase mine:
https://askubuntu.com/questions/1264568/increase-swap-in-20-04

**Reply by MWRtv** [reply]
> As for your serverIP, you need to input your external IP or people outside of your home network won't be able to join. As for the IPTables, that depends on your Linux version and if it's using IPTables. As for the swap, you can change it again, depending on what flavor your running. I'm using Ubuntu 20.04, and followed this guide to increase mine: https://askubuntu.com/questions/1264568/increase-swap-in-20-04

To clarify about the external IP, this is the one that shows up on the server list in-game for the IP address section when the server is highlighted(when it is not unlisted)? And would it be the same IP that would appear if I did a IP address lookup online?

I did a quick search and my OS does use `iptables` (in general), but I'm not sure if it needs or turned on or activated, or if it only needs rules to be added. It also has the `ufw` command (which I used to allow the ports for the server), so I'm confused whether I have to use both.

I'm running Parrot OS, it's build on Debian, which is what Ubuntu is based off of iirc. Why I have Parrot OS? A friend recommended me to use it, so I tried it (and haven't switched back to a normal flavour). However it's a little bit in need of an upgrade, and it has a weird command to update/upgrade it (should be `apt-get update && apt-get upgrade`, but it's not for this flavour of linux. I've tried it using the Debian commands but it broke the system last time I tried). I'm thinking of just doing a barebones install of Debian and setting up the sudo user and other user accounts needed for the SteamCMD/server files. But, before that I will try the link you posted. 

Thanks for the quick reply, btw.

@LordSith77 Were you able to get your server for 7dtd up and running?

**Reply by LordSith77** [reply]
I did eventually get it going. My problem was that my server wasn't using UFW at all (I use VirtualMin on this server), and hadn't opened the ports in the firewall (since I was thinking it was UFW and running commands for those). I have since changed even that over to CSF, which from my understanding is just so much better. I had everything up and running with normal install though, not using GSM. 
Of course now that A21 is out, I'm running into all sorts of new issues. But That's 7DtD for you. LOL

**Comment by KeepSmileHope** [comment]
Thanx for your answer, the fault was in the settings data under disable
networking. For what ever reason it seems that
7dtd had blocked steam networking by itself. How ever if you clean out the
"steam networking" the connection was fine.

BadPlayer ***@***.***> schrieb am Mi., 30. März 2022, 22:47:

**Reply by XBadPlayerX** [reply]
But....you're not the original poster??

**Comment by KeepSmileHope** [comment]
Dunno i got the Mail :o so i thought you mean me. Sorry for interrupting


BadPlayer ***@***.***> schrieb am Do., 31. März 2022, 14:16:
